public/pages/DetectorDetail/utils/helpers.tsx

```diff
@@ -142,11 +142,22 @@ export const getDetectorStateDetails = (
   );
 };
 
+/**
+ * Checks if any of the given indices contain the specified index as a prefix.
+ *
+ * This function iterates through an array of `CatIndex` objects and checks if the `index` property of any
+ * `CatIndex` object starts with the specified `index` string. It returns `true` if such an `index` is found,
+ * otherwise it returns `false`.
+ *
+ * @param index - The prefix string to check against the `index` properties of the `CatIndex` objects.
+ * @param indices - An array of `CatIndex` objects to search through.
+ * @returns A boolean value indicating whether any `CatIndex` object's `index` property starts with the specified prefix.
+ */
 export const containsIndex = (index: string, indices: CatIndex[]) => {
   let containsIndex = false;
   if (!isEmpty(indices)) {
     indices.forEach((catIndex: CatIndex) => {
-      if (get(catIndex, 'index', '') == index) {
+      if (get(catIndex, 'index', '').startsWith(index)) {
```

Comment:
Could this give an issue where another unrelated index exists that has the same prefix with other additional characters but its not used for AD

Reply:
Yes, it is possible. For example, customer configured opensearch-ad-result-a, but customer has another index opensearch-ad-result-ab.  Index opensearch-ad-result-ab-history-0607-1 exists, but opensearch-ad-result-a-history-0607-1 does not. I will change to check for alias.